### ISSUE TITLE
`patrol develop` doesn't exit on SIGINT and Hot Restart with "R" requires confirming with `Enter` key [dev]

### ISSUE THREAD

**Issue by zoskar** [leancodepl/patrol#1365]
### Steps to reproduce

1. Run `patrol develop` for any test
2. Wait until test finishes regardless of the result
3. try to use hot reload feature by typing `r` key
4. See no reaction
5. Press `return` key to see hot reload
6. wait to the end of the test
7. try to close Patrol process with `ctrl+c`
8. See ` App shut down on request` with no actual reaction
9. repeat step 7.
10. See error

### Actual results

Hot reloading requires confirmation by pressing `return` key which is not the case for Android emulators. After trying to quit `patrol develop` there is no reaction besides `App shut down on request` line. Second try to close results in error:
```
error while disposing
Bad state: This dispose scope is already disposed
#0      DisposeScope.dispose (package:dispose_scope/src/dispose_scope.dart:31:7)
#1      PatrolCommandRunner.dispose (package:patrol_cli/src/runner/patrol_command_runner.dart:219:27)
#2      patrolCommandRunner.<anonymous closure> (package:patrol_cli/src/runner/patrol_command_runner.dart:57:18)
#3      _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
#4      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#5      _DelayedData.perform (dart:async/stream_impl.dart:515:14)
#6      _PendingEvents.handleNext (dart:async/stream_impl.dart:620:11)
#7      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:591:7)
#8      _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#9      _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
#10     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:123:13)
#11     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:190:5)
```

### Logs

<details>
<summary>Logs</summary>

```
• Running app with entrypoint complete_poll_single_test.dart for iOS simulator on simulator iPhone 14 Pro Max...

        The Xcode project defines schemes: dev, prod, tst
        You must specify a --flavor option to select one of the available schemes.
Hot Restart: app connected (press "r" to restart)
^C✓ App shut down on request (397.6s)
^Cerror while disposing
Bad state: This dispose scope is already disposed
#0      DisposeScope.dispose (package:dispose_scope/src/dispose_scope.dart:31:7)
#1      PatrolCommandRunner.dispose (package:patrol_cli/src/runner/patrol_command_runner.dart:219:27)
#2      patrolCommandRunner.<anonymous closure> (package:patrol_cli/src/runner/patrol_command_runner.dart:57:18)
#3      _RootZone.runUnaryGuarded (dart:async/zone.dart:1594:10)
#4      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:339:11)
#5      _DelayedData.perform (dart:async/stream_impl.dart:515:14)
#6      _PendingEvents.handleNext (dart:async/stream_impl.dart:620:11)
#7      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:591:7)
#8      _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#9      _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
#10     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:123:13)
#11     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:190:5)
```

</details>

### Patrol version

  patrol: 2.0.0-dev.4

### Patrol Doctor output

<details>
<summary>Patrol Doctor output</summary>

```
Program adb found in /Users/oskarzajac/Library/Android/sdk/platform-tools/adb
Env var $ANDROID_HOME is not set
Program xcodebuild found in /usr/bin/xcodebuild
Program ideviceinstaller found in /opt/homebrew/bin/ideviceinstaller
```

</details>

### Flutter Doctor output

<details>
<summary>Flutter Doctor output</summary>

```
oskarzajac@DESKTOP-1KV7Q59 mobile % flutter doctor -v
[✓] Flutter (Channel stable, 3.10.0, on macOS 12.6 21G115 darwin-arm64, locale en-GB)
    • Flutter version 3.10.0 on channel stable at /Users/oskarzajac/Documents/tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (5 weeks ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/oskarzajac/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • ANDROID_SDK_ROOT = /Users/oskarzajac/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] VS Code (version 1.79.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.66.0

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 2BEABA86-73FB-464E-820C-D90F5A611372 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-2 (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 114.0.5735.133

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by bartekpacia** [comment]
Just to confirm: this only occurs on iOS, right?

**Comment by zoskar** [comment]
Yes, only for iOS

**Comment by bartekpacia** [comment]
Couldn't reproduce it with the Patrol example app.

patrol 2.0.0-dev.6
patrol_cli 2.0.0-dev.2

**Comment by bartekpacia** [comment]
Reproduced both issues with the customer app.

- `r` without `<Enter>` doesn't work
- SIGINT does not make `patrol_cli` exit

**Comment by bartekpacia** [comment]
Fixed in #1377.

**Comment by bartekpacia** [comment]
Fixed in #1377.
